gui.py

```diff
@@ -0,0 +1,401 @@
+            self._summary()
+            self._book_settings()
+            self._database_settings()
+            self._move_selection_settings()
+            self._engine_settings()
+
+    def _menu_bar(self):
+        def reload_settings_and_restart_gui():
+            self.settings.load_from_file()
+            # the easiest way to show reloaded settings is to destroy the gui and recreate it
+            dpg.delete_item(PRIMARY_WINDOW_TAG)
+            self._create_primary_window()
+
+        with dpg.menu_bar():
+            with dpg.menu(label="Settings"):
+                dpg.add_menu_item(label="Load", callback=reload_settings_and_restart_gui)
+                dpg.add_menu_item(label="Save", callback=lambda: self.settings.save_to_file())
+
+            with dpg.menu(label="Help"):
+                _menu_link("Announcement blog post and FAQ", BLOG_LINK)
+                _menu_link("Source code", SOURCE_CODE_LINK)
+
+    def _summary(self):
+        s = self.settings
+        dpg.add_text("An automatic practical chess opening repertoire builder using Lichess opening explorer API")
+        dpg.add_text("Customize your settings and then press the button below to begin generating your repertoire")
+        status = GenerationStatus()
+
+        def get_invalid_books() -> List[Book]:
+            invalid_books = list()
+            for book in s.book.get_books():
+                if not book.is_valid_pgn():
+                    invalid_books.append(book)
+            return invalid_books
+
+        def start_generation(button_tag):
+            # validate engine path
+            if s.engine.enabled:
+                if s.engine.path == s.engine.NO_FILE_SELECTED:
+                    status.error("No engine path was provided",
+                                 "Provide it under 'Engine settings' or unselect 'Use engine' and retry")
+                    return
+                if not os.path.exists(s.engine.path):
+                    status.error(f"Provided engine path '{s.engine.path}' does not exist\n",
+                                 "Correct it under 'Engine settings' or unselect 'Use engine' and retry")
+                    return
+
+            # validate books from free-text input are valid
+            invalid_books = get_invalid_books()
+            if len(invalid_books) > 0:
+                status.error("Book(s) listed below are invalid. Correct them under 'Book settings' and retry\n",
+                             "\n\n".join([book.__str__() for book in invalid_books]))
+                return
+
+            def finish_callback():
+                dpg.enable_item(button_tag)
+                status.info("Finished generating your repertoire",
+                            "You will find your PGNs in the same folder where BookBuilder is located")
+
+            status.info("PGN generation started")
+            dpg.disable_item(button_tag)
+            self.grower.run(self.settings, status, finish_callback)
```

Comment:
This is where the old main logic is actually called from - at the end of `start_generation()` method, after some basic validations.